### DISCUSSION TITLE
NOUS A5T Buttons do not work on Tasmota 14.2.0

### DISCUSSION THREAD

**Discussion by ToniH1987** [arendst/Tasmota#22066] [Q&A] [answered]
I've got two of the A5T and with Tasmota 14.1.0 everything works fine.
If I update to 14.2.0 everything works except the buttons for turning on/off the outlets.
The Rules are correct and enabled (buttons work as voltage dividers for the analog input).
I returned to 14.1.0 and it works again - on both Powerstrips.
So I suspect it's a problem with the version. But I'm very unsure if I should open an Issue.

**Comment by barbudor** [comment]
Strange
I have a [Gosund P1](https://templates.blakadder.com/gosund_P1.html) which is exactly the same hardware as the [Nous A5T](https://templates.blakadder.com/nous_A5T.html) and I'm running Tasmota 14.2.0.2 without any issues. Can you try the latest development version instead of the release ?

**Reply by ToniH1987** [reply]
Tried it with current dev build: 14.2.0.3 (d1bd31b-tasmota) 
Doesn't work. Back to 14.1.0 - works.
btw: The USB-Button works on all three versions.

Info says:
ESP Chip Id	14431124 (ESP8285N08)
Flash Chip Id	0x144051 (DOUT)

**Reply by barbudor** [reply]
Could you please report
- your current rules
- the different values of the analog input when buttons are not pressed and for each button pressed

**Reply by sfromis** [reply]
There was some recent enhancements around analog handling, hence the output from commands `AdcParam` and `AdcGpiio` could also be good to check.

**Reply by ToniH1987** [reply]
On 14.1:
{"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":34,"Free":477,"Rules":"ON Button1#State DO Power4 2 ENDON"}}
{"Rule2":{"State":"ON","Once":"ON","StopOnError":"OFF","Length":150,"Free":361,"Rules":"ON Analog#A0div10<30 DO Power3 2 BREAK ON Analog#A0div10<60 DO Power2 2 BREAK ON Analog#A0div10<80 DO Power1 2 ENDON BACKLOG RULE1 1; RULE2 1; RULE2 5"}}
{"Rule3":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":203,"Free":308,"Rules":"ON Energy#Power>%var1% DO Backlog var1 %value%; TelePeriod 3600; var2 %value%; add1 10; sub2 10 ENDON ON Energy#Power<%var2% DO Backlog var2 %value%; TelePeriod 3600; var1 %value%; add1 10; sub2 10 ENDON"}}

{"AdcParam1":[1,0,0,0.0005,0]}
AdcGpio - command unknown

Analog1 Values - Button 1: 741; Button 2: 465; Button 3: 217

On 14.2:
Identical to 14.1
 
 {"AdcParam1":[17,0,0,5,0]}
 {"AdcGpio":[0,0,5,0]}

Analog1 Values - Button 1: 740; Button 2: 465; Button 3: 215
(But I guess those slight deviations don't matter.)

**Reply by ToniH1987** [reply, marked as answer]
Got it:
Has to be:
{"AdcGpio":[0,1024,5,0]}

**Reply by HofeBY** [reply]
@DanielH1987
can you show us the Infosite of Tasmota ?

**Reply by sfromis** [reply]
While the AdcParam output did indeed show the problem, I suspect that it may have been broken by downgrading to a previous version, which is not an operation supported by Tasmota.

**Reply by ToniH1987** [reply]
On those two devices I only ever upgraded versions. Always the release, with the exception of the dev version earlier because it was suggested.

**Reply by ToniH1987** [reply]
Program Version	14.2.0 (release-tasmota)
Build Date & Time	2024-08-14T12:36:35
Core/SDK Version	2_7_7/2.2.2-dev(38a443e)
Uptime	0T03:20:16
Flash Write Count	3931 at 0xF4000
Boot Count	104
Restart Reason	Software/System restart
Friendly Name 1	<redacted>
Friendly Name 2	<redacted>
Friendly Name 3	<redacted>
Friendly Name 4	<redacted>
AP2 Information	SSId	<redacted>
RSSI 68%, -66 dBm
Mode 11n
Channel 1
BSSId 	<redacted>
Hostname		<redacted>
MAC Address		<redacted>
IP Address (WiFi)		<redacted>
Gateway		<redacted>
Subnet Mask	255.255.255.0
DNS Server1		<redacted>
DNS Server2	0.0.0.0
HTTP API	Disabled
MQTT Host		<redacted>
MQTT Port	1883
MQTT User		<redacted>
MQTT Client		<redacted>
MQTT Topic		<redacted>
MQTT Group Topic 1	cmnd/tasmotas/
MQTT Full Topic	cmnd/<redacted>
MQTT Fallback Topic	cmnd/<redacted>
MQTT No Retain	Disabled
Emulation	None
ESP Chip Id	14431124 (ESP8285N08)
Flash Chip Id	0x144051 (DOUT)
Flash Size	1024 KB
Program Flash Size	1024 KB
Program Size	648 KB
Free Program Space	352 KB
Free Memory	24.9 KB

**Reply by HofeBY** [reply]
thx,
so I know which flash size I have to backup, before I flash a changed funktional TM one.
Because I have a "Nous A5" usable with "Nous smart" currently without Tasmota.

**Reply by sfromis** [reply]
(You mentioned upgrading to 14.2.0 and then "returned to 14.1.0" which constitutes a downgrade. The point is that any upgrade *might* do things which an older version might change, to be different from what it had been)

**Comment by ToniH1987** [comment]
As soon as I upgraded from 14.1 to 14.2 for the first time the issue existed.
Maybe that wasn't clear...

**Reply by sfromis** [reply]
To be clear, what I meant was more that the output from `AdcParam` after the downgrade + re-upgrade might not be the same as after the first upgrade. While this may not be a causal factor, it might influence how the specifics are visible now. What you posted showed (as you noticed) that something was broken, where the circumstances of this may be "more interesting" than if it worked or not, when it comes to looking for, and fixing, an issue.
